packages/scene-composer/src/components/panels/scene-settings/ComponentVisibilityToggle.tsx

```diff
@@ -0,0 +1,64 @@
+import React, { useContext, useMemo } from 'react';
+import { useIntl } from 'react-intl';
+import { Box, Toggle } from '@awsui/components-react';
+
+import { IDataOverlayComponentInternal, useStore, useViewOptionState } from '../../../store';
+import { sceneComposerIdContext } from '../../../common/sceneComposerIdContext';
+import { KnownComponentType } from '../../../interfaces';
+import { Component } from '../../../models/SceneModels';
+import { findComponentByType } from '../../../utils/nodeUtils';
+
+export interface ComponentVisibilityToggleProps {
+  componentType: KnownComponentType | Component.DataOverlaySubType;
+  label: string;
+  onChange?: (newValue: boolean) => void;
+}
+
+export const ComponentVisibilityToggle: React.FC<ComponentVisibilityToggleProps> = ({
+  componentType,
+  label,
+  onChange,
+}) => {
+  const sceneComposerId = useContext(sceneComposerIdContext);
+  const toggleComponentVisibility = useViewOptionState(sceneComposerId).toggleComponentVisibility;
+  const componentVisible = useViewOptionState(sceneComposerId).componentVisibilities[componentType];
+  const getComponentRefByType = useStore(sceneComposerId)((state) => state.getComponentRefByType);
+  const getSceneNodeByRef = useStore(sceneComposerId)((state) => state.getSceneNodeByRef);
+  const componentNodeMap = useStore(sceneComposerId)((state) => state.document.componentNodeMap);
+  const { formatMessage } = useIntl();
+
+  const hasComponent = useMemo(() => {
```

Comment:
What this method signifies?

Reply:
It's returning the component based on whether it's a  `Component.DataOverlaySubType` or `KnownComponentType`.

Reply:
this is to check whether the component exists. when there is no instance of the specified component type, the toggle will be disabled.